### PR TITLE
Provide the CloseWindow warning experience for the 'X' button

### PR DESCRIPTION
**Pull request by KaiyuWang16**
This is a follow up project of #1589 , which warns users of multiple tabs opened when press ALT+F4. 
Now we are going to connect this experience with the 'X' button of the window. 

For now, this PR only includes the implementation for IslandWindow (client window). Non-client window experience is still under implementation. 

 
## PR Checklist
* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments

To Test:

1. Start terminal. 
2. Open multiple tabs. 
3. Click 'X' on the top right corner of the window. 
4. You should be able  to see a warning. If you cancel it, then the Terminal keeps running. 
5. If you click 'Close', then the app will close. 
6. If you only have one tab opened, and click 'X', the app should close directly without any warnings.

### CODE REVIEW

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 1]
```diff
@@ -69,6 +69,12 @@ void AppHost::Initialize()
         // content in Create.
         _app.SetTitleBarContent({ this, &AppHost::_UpdateTitleBarContent });
     }
+    else
+    {
+        // Register the 'X' button of the window for a warning experience of multiple
+        // tabs opened, this is consistent with Alt+F4 closing
+        _window->WindowCloseButtonClicked([this]() { _app.WindowCloseButtonClicked(); });
+    }
```
[the remark this answers is not in the document]
Comment (DHowett-MSFT) [reply src/cascadia/WindowsTerminal/AppHost.cpp]: That's a good question. It seems like NCIslandWindow will be able to leverage the same WM_CLOSE event, *but* the `MinMaxCloseControl` will need to send the right SYSCOMMAND to make that happen.

[the remark this answers is not in the document]
Comment (KaiyuWang16) [reply src/cascadia/WindowsTerminal/AppHost.cpp]: Sorry for the misunderstanding. NonClientIslandWindow is not working for now, the current PR only includes the changes for Client Island window. I am implementing the close experience for Client island now.

[the remark this answers is not in the document]
Comment (KaiyuWang16) [reply src/cascadia/WindowsTerminal/AppHost.cpp]: Now the close experience work for both client and non-client island window.

**Comment by DHowett-MSFT** [comment]
_please make sure your pull request names are reasonable_ ("dev/kawa/1589  do the thing" is not a great branch name)

Pretend that they are completing the sentence, "This pull request will ..."

**Comment by KaiyuWang16** [comment]
> please make sure your pull request names are reasonable ("dev/kawa/1589  do the thing" is not a great branch name)
> Pretend that they are completing the sentence, "This pull request will ..."

PR name changed. I wonder if I need to change the branch name ? This may require to post a new PR.

**Comment by DHowett-MSFT** [comment]
No, you can keep the branch name.

**Pull request merged**